standards.md

```diff
@@ -0,0 +1,315 @@
+
+<p>&nbsp;</p>
+
+### Contents file <a name="contents"></a>
+
+If there are multiple configurations within the DEM, there must be a JSON file named “contents.json” that contains the module name and a list of the settings files within the DEM.
+
+The required base level keys of the settings file are as follows:
+| Key  | Definition |
+|------|------------|
+| module | Provides the DEM name. |
+| contents | List of all DEM settings file names (relative to DEM directory). |
+
+<p>&nbsp;</p>
+
+The following module keys are required:
+| Key  | Definition |
+|------|------------|
+| name | Short name of DEM |
+| long_name | A human readable name of module |
+
+<p>&nbsp;</p>
+
+## Runtime Environment <a name="runtimeenvironment"></a>
+
+This section describes the runtime environment, as initialized by the framework. When executed through the CMEC/MDTF interface, the DEMs may assume that all conditions below are met.
+
+All executables and third-party libraries requested in the settings file will be available on the $PATH seen by the DEM.  
+
+The following environment variables will be set:
+
+| Key  | Definition |
+|------|------------|
+| CMEC_CODE_DIR <br>MDTF_POD_CODE_DIR| The directory where the DEM’s code can be found |
+| CMEC_OBS_DATA<br>MDTF_POD_OBS_DATA| The directory where the relevant observational data can be found. |
+| CMEC_MODEL_DATA<br>MDTF_MODEL_DATA | The directory containing the model input data.  If autocurated metadata has been generated, it can be found in the file autocurator.json in the root directory. |
+| CMEC_WK_DIR<br>MDTF_POD_WK_DIR | The working directory for output from the DEM. |
+| MDTF_CASENAME<br>MDTF_FIRSTYR<br>MDTF_LASTYR | (optional) Variables used by MDTF PODs for labeling plots. |
+| CMEC_DEBUG<br>MDTF_DEBUG | (optional) If set to 1 then the DEM should output more verbose logging information to help with debugging. |
+| CMEC_SAVE_NC<br>MDTF_SAVE_NC | (optional) If set to 0 then the DEM should skip any steps where raw output data is saved in netCDF format. |
+
+<p>&nbsp;</p>
+
+The following restrictions hold on DEM operation:  
+
+DEMs should not refer to, or attempt to access, any files outside of the directories specified by the environment variables above.
+All files in *_CODE_DIR, *_OBS_DATA, or *_MODEL_DATA should be treated as read-only.  
+
+Output from DEMs should only be written to *_WK_DIR. Any intermediate files not part of the output bundle should be deleted before exiting. DEMs should not assume persistence of any files across runs.  
+
+Temporary files, if needed, can be written to the working directory, preferably in the “temp” subfolder. This directory should be removed once execution is completed successfully.  
+
+As a best practice, DEMs should not attempt to open network connections or access resources on remote hosts.  
```

Comment:
I wonder if this should be a requirement as opposed to a best-practice.  Remote connections will not be allowed in operational settings.

Reply:
@jkrasting given how many operational platforms would not allow this, it makes sense to make it a requirement.